test/e2e/mv3/multiple-restarts.spec.js

```diff
@@ -0,0 +1,385 @@
+    }
+  });
+
+  it('Should lock wallet when a browser session ends (after turning off the extension', async function () {
+    await withFixtures(
+      {
+        dapp: true,
+        fixtures: new FixtureBuilder()
+          .withPermissionControllerConnectedToTestDapp()
+          .build(),
+        ganacheOptions: generateGanacheOptions({
+          concurrent: { port: 8546, chainId: 1338 },
+        }),
+        title: this.test.title,
+      },
+      async ({ driver }) => {
+        const { extensionUrl } = driver;
+        const extensionId = extensionUrl.split('//')[1];
+
+        await driver.navigate();
+
+        await unlockWallet(driver, WALLET_PASSWORD);
+
+        await reloadExtension(driver, extensionId);
+
+        // ensure extension finishes reloading before reopening full screen extension
+        await sleepSeconds(0.1);
+
+        await driver.openNewPage(`${extensionUrl}/home.html`);
+
+        const passwordField = await driver.isElementPresent('#password');
+        assert.ok(
+          passwordField,
+          'Password screen is not visible. Wallet should have been locked.',
+        );
+      },
+    );
+
+    async function reloadExtension(driver, extensionId) {
+      await switchToWindow(driver, WINDOW_TITLES.ExtensionInFullScreenView);
+
+      await driver.openNewPage('chrome://extensions/');
+
+      // extensions-manager
+      const extensionsManager = await driver.findElement('extensions-manager');
+
+      // shadowRoot
+      const extensionsManagerShadowRoot = await driver.executeScript(
+        'return arguments[0][0].shadowRoot',
+        extensionsManager,
+      );
+
+      // cr-view-manager
+      const viewManager = await extensionsManagerShadowRoot.findElement({
+        css: '#viewManager',
+      });
+
+      // extensions-item-list
+      const itemList = await viewManager.findElement({
+        css: '#items-list',
+      });
+
+      // shadowRoot
+      const itemListShadowRoot = await driver.executeScript(
+        'return arguments[0][0].shadowRoot',
+        itemList,
+      );
+
+      // extension-item
+      const extensionItem = await await itemListShadowRoot.findElement({
+        css: `#${extensionId}`,
+      });
+
+      // shadowRoot
+      const extensionItemShadowRoot = await driver.executeScript(
+        'return arguments[0][0].shadowRoot',
+        extensionItem,
+      );
+
+      // cr-icon-button
+      const devReloadButton = await extensionItemShadowRoot.findElement({
+        css: '#dev-reload-button',
+      });
+
+      // shadowRoot
+      const devReloadButtonShadowRoot = await driver.executeScript(
+        'return arguments[0][0].shadowRoot',
+        devReloadButton,
+      );
+
+      // cr-icon-button
+      const reloadBtn = await devReloadButtonShadowRoot.findElement({
+        css: '#maskedImage',
+      });
+
+      await reloadBtn.click();
+    }
+  });
+});
+
+async function unlockWallet(driver, walletPassword) {
```

Comment:
I found those functions helpful and used in other tests, we could move them out in this ticket and encourage the reuse in the future?